cmd/tools/testwick/testwicker.go

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+//
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"time"
+
+	cmodel "github.com/mattermost/mattermost-cloud/model"
+	mmodel "github.com/mattermost/mattermost-server/v5/model"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// TestWicker data struct for test wicker scenarios
+type TestWicker struct {
+	provisionerClient *cmodel.Client
+	mmClient          *mmodel.Client4
+	logger            *logrus.Logger
+	installation      *cmodel.InstallationDTO
+	userID            string
+	channelID         string
+	teamID            string
+	Error             error
```

Comment:
I am not sure about error handling here. Let's consider the following scenario:
- `CreateInstallation` failes
- `Error` is set to some value but the flow continues
- `WaitForInstallationStable` will also fail as the installation was not created
- It will override `Error` and the process will continue most likely failing further along the way

If I am not missing anything this will cause only the last error to be logged therefore it may be hard to troubleshoot.

Reply:
Yeah this is true, I think I can split it to `testwicker, error`

Reply:
@Szymongib what if we just keep a list of errors and dump them as logs if `len(w.Error) > 0`. Trying to think how we can keep the declarative usage but still keep all the logs.

Reply:
Or even better have a `Runner()` which gets as input functions and iterate through them:

```go
runner.Execute([func1, func2])
```

internally the runner checks for errors and fails accordingly so we can avoid the multiple `if err!=nil` for every step

Reply:
That sounds good, I think we actually may have something similar in e2e tests https://github.com/mattermost/mattermost-cloud/blob/master/e2e/workflow/engine.go#L30